src/components/Calculator.js

```diff
@@ -1,30 +1,106 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Calculator.css';
 import DigitButton from './DigitButton';
+import operate from './logic/operate';
 
-const Calculator = () => (
-  <div className="calculator-grid">
-    <div className="calculator-output">123</div>
-    <button type="button">AC</button>
-    <button type="button">+/-</button>
-    <button type="button">%</button>
-    <button type="button" className="yellow-btns">÷</button>
-    <DigitButton digit={7} />
-    <DigitButton digit={8} />
-    <DigitButton digit={9} />
-    <button type="button" className="yellow-btns">x</button>
-    <DigitButton digit={4} />
-    <DigitButton digit={5} />
-    <DigitButton digit={6} />
-    <button type="button" className="yellow-btns">-</button>
-    <DigitButton digit={1} />
-    <DigitButton digit={2} />
-    <DigitButton digit={3} />
-    <button type="button" className="yellow-btns">+</button>
-    <button type="button" className="span-tow">0</button>
-    <button type="button">.</button>
-    <button type="button" className="yellow-btns">=</button>
-  </div>
-);
+const Calculator = () => {
+  const [result, setResult] = useState('0');
+  const [input, setInput] = useState('');
+  const [operator, setOperator] = useState('');
+
+  const handleOperatorClick = (value) => {
+    if (operator === '') {
+      setInput(result);
+      setResult('0');
+      setOperator(value);
+    } else if (operator !== '=' && result !== '0') {
+      const newResult = operate(input, result, operator);
+      setInput(newResult);
+      setResult('0');
+      setOperator(value);
+    } else if (operator === '=') {
+      setInput(result);
+      setResult('0');
+      setOperator(value);
+    } else if (value === '%') {
+      if (result === '0') {
+        const percentage = operate(input, '1', operator);
+        setInput(input);
+        setResult(percentage);
+        setOperator(value);
+      } else {
+        const percentage = operate(result, '100', operator);
+        setInput(result);
+        setResult(percentage);
+        setOperator(value);
+      }
```

Comment:
- [ ] This is supposed to work as a modulus, not a percentage, so when I input 7 % 2 I should get 1 which is a modulus of 7 and 2, currently, I get 0.07 kindly fix this